src/super_context.ml

```diff
@@ -486,7 +487,10 @@ let dump_env t ~dir =
   Ocaml_flags.dump (Env.ocaml_flags t ~dir)
 
 let resolve_program t ?hint ~loc bin =
-  Artifacts.binary ?hint ~loc t.artifacts bin
+  Hashtbl.find_or_add
+    t.artifacts_cache
+    ~f:(Artifacts.binary ?hint ~loc t.artifacts)
```

Comment:
I'm for the cache, but I think the loc and the hint should be part of the hash. With the current cache, we'll always be reporting the first loc encountered which isn't really correct.

Reply:
Actually, I think it's better this way. For instance, if refmt is missing, there is no point reporting the error several times at different locations.